pkg/logging/logging.go

```diff
@@ -37,5 +37,9 @@ const (
 )
 
 func LogStatsEntries(client *constraintclient.Client, logger logr.Logger, entries []*instrumentation.StatsEntry, msg string) {
+	if len(entries) == 0 {
```

Comment:
I think it's useful to skip logging empty stats as they provide no information. This is a cheap way to do so. The main advantage here is that I have found the logging to be too spurious for all the GVKs that don't necessarily get "reviewed" (where "review" = an eval call to the opa engine is made).

It's mostly `Lease` resources:

``` json
{
  "level": "info",
  "ts": 1684281683.4371057,
  "logger": "webhook",
  "msg": "admission review request stats",
  "hookType": "validation",
  "process": "admission",
  "event_type": "review_response_stats",
  "resource_group": "coordination.k8s.io",
  "resource_api_version": "v1",
  "resource_kind": "Lease",
  "resource_namespace": "kube-system",
  "request_username": "system:kube-scheduler",
  "execution_stats": null
}
```

I'm also open to inspecting the GVKs at the call site and bail if it's `Lease`. 

---

Looking at this from a different lens, if we do skip  logging on empty stats for ALL gvks, we run into the scenario where folks might expect the stats but they don't actually get logged (say bc stats collection is now broken in CF). It's a contrived example but it illustrates how this approach can be a papercut.

Reply:
Could we tie this with GVKs that are part of constraints? So only output for GVKs that are defined in policies by default to reduce the noise.

If it's omitted or `*`, we are back to this issue though.

Reply:
Not logging stats when match criteria fails vs. not logging empty stats are currently equivalent approaches, since match criteria failing causes the driver (currently the only source of stats) to not be invoked.

IMO it's okay to not log stats when there are no stats to log, however at some point generating stats based off match criteria could be interesting.

Long term, mechanisms for more targeted stats gathering will be interesting (e.g. "give me all stats relevant for X particular category of requests, which appear to be slow"), vs. dumping all stats, but the shape of that targeting mechanism is not immediately obvious.

Reply:
+1 on not logging empty stats.
generating stats based off match criteria could be very helpful for debugging bad match criteria set in the constraints.

Reply:
It sounds like there's agreement here that not logging empty stats is fine for now. I've added an issue to track adding match criteria stats in CF: https://github.com/open-policy-agent/frameworks/issues/328